library/script-library/src/main/java/uk/gov/gchq/gaffer/script/operation/platform/LocalDockerPlatform.java

```diff
@@ -0,0 +1,169 @@
+import uk.gov.gchq.gaffer.script.operation.image.DockerImage;
+import uk.gov.gchq.gaffer.script.operation.image.Image;
+import uk.gov.gchq.gaffer.script.operation.util.DockerClientSingleton;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class LocalDockerPlatform implements ImagePlatform {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RunScriptHandler.class);
+    private DockerClient docker = null;
+    private String dockerfilePath = "";
+    private int port;
+
+    /**
+     * Builds a docker image
+     *
+     * @param scriptName             the name of the script being run
+     * @param scriptParameters       the parameters of the script being run
+     * @param pathToBuildFiles       the path to the directory containing the build files
+     * @return the docker image
+     */
+    public DockerImage buildImage(final String scriptName, final Map<String, Object> scriptParameters, final String pathToBuildFiles) {
+
+        final DockerImageBuilder dockerImageBuilder = new DockerImageBuilder();
+
+        // Get the user defined dockerfile or use the default
+        dockerImageBuilder.getFiles(pathToBuildFiles, dockerfilePath);
+
+        // Connect to the Docker client. To ensure only one reference to the Docker client and to avoid
+        // memory leaks, synchronize this code amongst multiple threads.
+        LOGGER.info("Connecting to the Docker client...");
+        docker = DockerClientSingleton.getInstance();
+        LOGGER.info("Docker is now: {}", docker);
+        final DockerImage dockerImage = (DockerImage) dockerImageBuilder.buildImage(scriptName, scriptParameters, pathToBuildFiles);
+
+        // Remove the old images
+        final List<com.spotify.docker.client.messages.Image> images;
+        try {
+            images = docker.listImages();
+            String repoTag = "[<none>:<none>]";
+            for (final com.spotify.docker.client.messages.Image image : images) {
+                if (Objects.requireNonNull(image.repoTags()).toString().equals(repoTag)) {
+                    docker.removeImage(image.id());
+                }
+            }
+        } catch (final DockerException | InterruptedException e) {
+            LOGGER.info("Could not remove image, image still in use.");
+        }
+
+        return dockerImage;
+    }
+
+    /**
+     * Builds a docker image and creates a docker container instance.
+     *
+     * @param image                  the image to create a container from
+     * @param ip                     the ip the container is connected to
+     * @return the docker container
+     */
+    @Override
+    public Container createContainer(final Image image, final String ip) {
+
+        String containerId = "";
+        // Keep trying to create a container and find a free port.
+        try {
+            port = RandomPortGenerator.getInstance().generatePort();
```

Comment:
If this produces a port number that's being used by another process, the binding will fail, but the port will be being marked as used by the generator even though it didn't generate it as a free port to be used. Consider releasing the port if use by the container is unsuccessful. Also, perhaps checking to see whether the port is available at generation time and use time might be a good idea before trying to use.